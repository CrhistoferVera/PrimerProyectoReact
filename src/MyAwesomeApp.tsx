import type { CSSProperties } from "react";

const firstName = "Crhistofer";
const lastName = "Vera";

const favoriteGames = ["Fortnite", "Rust", "Valorant"];
const isActive = true;
const address = {
    zipCode: "abc123",
    country : "Bolivia"
}

const myStyles: CSSProperties= {
    backgroundColor: "#fafafa",
    borderRadius: 20,
    padding: 10,
    marginTop: 20
}
export function MyAwesomeApp(){

    return(
        <div>
            <h1 style={myStyles} data-testid="first-name-title">{firstName}</h1>
            <h3>{lastName}</h3>
            <p className="mi-clase-favorita">{favoriteGames.join(", ")}</p>
            <h1>{isActive ? "activo" :"inactivo"}</h1>
            <p
            style={{
                backgroundColor: "red",
                borderRadius: 10,
                padding: 10,
            }}>{ JSON.stringify(address)}</p>
        </div>
    );
}