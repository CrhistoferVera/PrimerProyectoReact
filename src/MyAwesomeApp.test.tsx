import { describe,expect,test } from "vitest";
import{render,screen} from '@testing-library/react'
import { MyAwesomeApp } from "./MyAwesomeApp";


describe('MyAwesomeApp', ()=>{
    test('should rendder firstName and lastName',()=>{
        const {container} = render(<MyAwesomeApp/>);

        render(<MyAwesomeApp/>);

        const h1 = container.querySelector('h1');
        expect(h1?.innerHTML).toContain("Crhistofer");

        const h3 = container.querySelector('h3');
        expect(h3?.innerHTML).toContain("Vera");
        


    });
    test('should rendder firstName and lastName - Screen',()=>{
        render(<MyAwesomeApp/>);
        screen.debug();

        //const h1 = screen.getByRole('heading',{
        //    level:1
        //});
        const h1 = screen.getByTestId("first-name-title");
        expect(h1.innerHTML).toContain("Crhistofer");
    });

    test("should match snapshot", ()=>{
        const {container}= render(<MyAwesomeApp/>);
        expect(container).toMatchSnapshot();
    })
});