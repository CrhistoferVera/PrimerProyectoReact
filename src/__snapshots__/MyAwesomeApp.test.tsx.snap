// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div>
    <h1
      data-testid="first-name-title"
      style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 20px;"
    >
      Crhistofer
    </h1>
    <h3>
      Vera
    </h3>
    <p
      class="mi-clase-favorita"
    >
      Fortnite, Rust, Valorant
    </p>
    <h1>
      activo
    </h1>
    <p
      style="background-color: red; border-radius: 10px; padding: 10px;"
    >
      {"zipCode":"abc123","country":"Bolivia"}
    </p>
  </div>
</div>
`;
